crates/client/mempool/src/block_production.rs

```diff
@@ -0,0 +1,159 @@
+//! This should probably be moved into another crate.
+
+use blockifier::blockifier::transaction_executor::TransactionExecutor;
+use blockifier::bouncer::{Bouncer, BouncerWeights, BuiltinCount};
+use blockifier::transaction::transaction_execution::Transaction;
+use dc_db::DeoxysBackend;
+use dc_exec::{ExecutionContext, BlockifierStateAdapter};
+use dp_utils::graceful_shutdown;
+use std::sync::Arc;
+
+use crate::{clone_account_tx, Error, Mempool};
+
+pub struct BlockProductionTask {
+    backend: Arc<DeoxysBackend>,
+    mempool: Arc<Mempool>,
+    executor: TransactionExecutor<BlockifierStateAdapter>,
+    current_pending_tick: usize,
+}
+
+const TX_CHUNK_SIZE: usize = 128;
+
+impl BlockProductionTask {
+    pub fn new(backend: Arc<DeoxysBackend>, mempool: Arc<Mempool>) -> Result<Self, Error> {
+        let mut pending_block = mempool.get_or_create_pending_block()?;
+        let mut executor = ExecutionContext::new(Arc::clone(&backend), &pending_block.info)?.tx_executor();
+
+        let bouncer_config = backend.chain_config().bouncer_config.clone();
+        executor.bouncer = Bouncer::new(bouncer_config);
+
+        Ok(Self { backend, mempool, executor, current_pending_tick: 0 })
+    }
+
+    fn update_pending_block(&mut self, finish_block: bool) -> Result<(), Error> {
+        let config_bouncer = &self.backend.chain_config().bouncer_config.block_max_capacity;
+
+        if finish_block {
+            // Full bouncer capacity
+            self.executor.bouncer.bouncer_config.block_max_capacity = config_bouncer.clone();
+        } else {
+            let current_pending_tick = self.current_pending_tick;
```

Comment:
We want ticks every two seconds - that makes it easier for optimistic parallelism later, and also it allows us to do writes to DB only every two seconds. I've heard (unsure?) this is also how the current starkware sequencer operates (- or at least that's how it makes sense to me)
Each of those tick store the pending block at the end.

Each of those tick have a fraction of the total bouncer capacity for the whole block.
Once the time has passed for the full block we complete it to the maximum bouncer capacity, and convert the pending block to a full block (ie. do commitment computations)

Reply:
This is not yet commented I think we shouldn't have put the PR in ready yet
I am moving things around quite a lot around here

Reply:
I might be repeating points from my comments above but 

I think we can check for filling blocks instead of a block time tick (we can have a very large one if needed but I assume that we will reach block capacity before that limit. this stems from our discussion in Brussels that we need longer time blocks to reduce # of blocks to reduce proving costs). We would need our block times to be dynamic based on the traffic.

> Each of those tick have a fraction of the total bouncer capacity for the whole block.
Why do we need this? If there were a 100 txs in 2 seconds that will fill the block, we are fine to do that right?

Reply:
> Why do we need this? If there were a 100 txs in 2 seconds that will fill the block, we are fine to do that right?

Then the blockchain isn't bounded in size, correct?